### DISCUSSION TITLE
0.6.0 Release Timeline

### DISCUSSION THREAD

**Discussion by adamjstewart** [torchgeo/torchgeo#2078] [Announcements]
I've gotten a lot of questions recently along the lines of:

* When is the next TorchGeo release going to be?
* How can I help contribute to TorchGeo?

I wanted to address both questions in a single place. We realize it's been a while since our last feature release, and we've accumulated a lot of great new datasets and models since v0.5.0. The v0.6.0 release is currently scheduled sometime in the next few weeks! If you are currently working on or planning a PR, now is the time to push it through. 

If you are interested in getting involved in TorchGeo development, the following features will be a big part of the v0.6.0 release:

1. Convert datasets from Radiant MLHub to Source Cooperative: https://github.com/microsoft/torchgeo/issues/1830
2. Adding new multi-modal geospatial foundation models: https://github.com/microsoft/torchgeo/issues/1043

We'll cut the v0.6.0 release as soon as 1 is complete (2 is unlikely to ever be "complete"). If you're interested in contributing, we're looking for people to help with both tasks. Simply comment on the issue and claim a dataset/model that you would like to add.

**Comment by adamjstewart** [comment]
Draft of the release notes:

# TorchGeo 0.6.0 Release Notes

TorchGeo 0.6 adds 15 new datasets and 27 new pre-trained models, encompassing 11 months of hard work by XXX contributors from around the world.

## Highlights of this release

### Multimodal foundation models

![Diagram of a unified multimodal Earth foundation model](https://github.com/user-attachments/assets/d231817f-9d48-49c2-ba79-56a2c20cc60e)

There are [thousands of Earth observation satellites](https://orbit.ing-now.com/) orbiting the Earth at any given time. Historically, in order to use one of these satellites in a deep learning pipeline, you would first need to collect millions of manually-labeled images from this sensor in order to train a model. Self-supervised learning enabled label-free pre-training, but still required millions of diverse sensor-specific images, making it difficult to use newly launched or expensive commercial satellites. 

TorchGeo 0.6 adds multiple new [multimodal foundation models](https://arxiv.org/abs/2405.04285) capable of being used with imagery from any satellite/sensor, even ones the model was not explicitly trained on. While [GASSL](https://arxiv.org/abs/2011.09980) and [Scale-MAE](https://arxiv.org/abs/2212.14532) only support RGB images, [DOFA](https://arxiv.org/abs/2403.15356) supports RGB, SAR, MSI, and HSI with any number of spectral bands. It uses a novel wavelength-based encoder to map the spectral wavelength of each band to a known range of wavelengths seen during training.

The following table describes the dynamic spatial (resolution), temporal (time span), and/or spectral (wavelength) support, either via their training data (implicit) or via their model architecture (explicit), offered by each of these models:

| Model | Spatial | Temporal | Spectral |
| --- | --- | --- | --- |
| [DOFA](https://arxiv.org/abs/2403.15356) | implicit | - | explicit |
| [GASSL](https://arxiv.org/abs/2011.09980) | implicit | - | - |
| [Scale-MAE](https://arxiv.org/abs/2212.14532) | explicit | - | - |

TorchGeo 0.6 also adds multiple new unimodal foundation models, including [DeCUR](https://arxiv.org/abs/2309.05300) and [SatlasPretrain](https://arxiv.org/abs/2211.15660).

### Source Cooperative migration

![source-migration](https://github.com/user-attachments/assets/b1328fde-5424-4ca5-af15-1ebb9cf2179c)

TorchGeo contains a number of datasets from the recently defunct [Radiant MLHub](https://mlhub.earth/):

* [AgriFieldNet Competition Dataset](https://beta.source.coop/radiantearth/agrifieldnet-competition/)
* [Smallholder Cashew Plantations in Benin](https://beta.source.coop/technoserve/cashews-benin/)
* [Sentinel-2 Cloud Cover Segmentation Dataset](https://beta.source.coop/radiantearth/cloud-cover-detection-challenge/)
* [CV4A Kenya Crop Type Competition](https://beta.source.coop/radiantearth/african-crops-kenya-02/)
* [Tropical Cyclone Wind Estimation Competition](https://beta.source.coop/nasa/tropical-storm-competition/)
* [Marine Debris Dataset for Object Detection in Planetscope Imagery](https://beta.source.coop/nasa/marine-debris/)
* [Rwanda Field Boundary Competition Dataset](https://beta.source.coop/nasa/rwanda-field-boundary-competition/)
* [South Africa Crop Type Competition](https://beta.source.coop/radiantearth/south-africa-crops-competition/)
* [SpaceNet Datasets](https://spacenet.ai/datasets/)
* [Western USA Live Fuel Moisture](https://beta.source.coop/stanford/sar-moisture-conent/)

These datasets were recently migrated to [Source Cooperative](https://source.coop/) (and AWS in the case of SpaceNet), but with a completely different file format and directory structure. It took a lot of effort, but we have finally ported all of these datasets to the new download location and file hierarchy. As an added bonus, the new data loader code is significantly simpler, allowing us to remove 2.5K lines of code in the process!

### OSGeo community project

![OSGeo logo](https://raw.githubusercontent.com/OSGeo/osgeo/master/marketing/branding/logo/osgeo-logo-cmyk.svg)

TorchGeo is now officially a member of the OSGeo community! [OSGeo](https://www.osgeo.org/) is a not-for-profit foundation for open source geospatial software, providing financial, organizational, and legal support. We are in good company, with other [OSGeo projects](https://www.osgeo.org/projects/) including GDAL, PROJ, GEOS, QGIS, and PostGIS. Membership in OSGeo promotes advertising of TorchGeo to the community, and also ensures that we follow best practices for the stability, health, and interoperability of the open source geospatial ecosystem.

All TorchGeo users are encouraged to join us on Slack, join our Hugging Face organization, and join us in OSGeo using any of the following badges in our README:

[![slack](https://img.shields.io/badge/Slack-join-4A154B?logo=slack)](https://join.slack.com/t/torchgeo/shared_invite/zt-22rse667m-eqtCeNW0yI000Tl4B~2PIw)
[![huggingface](https://img.shields.io/badge/Hugging_Face-join-FFD21E?logo=huggingface)](https://huggingface.co/torchgeo)
[![osgeo](https://img.shields.io/badge/OSGeo-join-4CB05B?logo=osgeo)](https://www.osgeo.org/community/getting-started-osgeo/)

### Lightning Studios support

![Lightning AI logo](https://github.com/user-attachments/assets/639877b0-1f7e-4e85-906c-301f2e545999)

TorchGeo has always had a close collaboration with Lightning AI, including active contributions to PyTorch Lightning and TorchMetrics. In this release, we added buttons allowing users to launch our tutorial notebooks in the new Lightning Studios platform. [Lightning Studios](https://lightning.ai/studios) is a more powerful version of Google Colab, with reproducible software and data environments allowing you to pick up where you left off, VS Code and terminal support, and the ability to quickly scale up to a large number of GPUs. All TorchGeo tutorials have been confirmed to work in both Lightning Studios and Google Colab, allowing users to get started with TorchGeo without having to invest in their own hardware.

**Comment by jodygarnett** [comment]
Nice! here is the osgeo community project logo: https://github.com/OSGeo/osgeo/tree/master/incubation/community

**Reply by adamjstewart** [reply]
Thanks! I knew there had to be an SVG somewhere...
